src/Rules/Classes/ClassConstantRule.php

```diff
@@ -39,6 +39,7 @@ public function __construct(
 		private RuleLevelHelper $ruleLevelHelper,
 		private ClassNameCheck $classCheck,
 		private PhpVersion $phpVersion,
+		private bool $checkNonStringableDynamicAccess = true,
```

Comment:
I understand that we shouldn't use the default value, but I was getting the following error, probably due to a missing DI configuration, so I temporarily added `= true` to test.

```
% make phpstan
php bin/phpstan clear-result-cache -q && php -d memory_limit=448M bin/phpstan

In Resolver.php line 677:

  Service 'rules.26' (type of PHPStan\Rules\Classes\ClassConstantRule): Parameter $checkNonStringableDynamicAccess in ClassConstantRule::__construct() has no class type or default value, so its value must be
  specified.
```

Reply:
My suggestion above will help you get rid of this error.